sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/Publishers.java

```diff
@@ -35,24 +36,28 @@
   private Publishers() {}
 
   @SuppressWarnings("unchecked")
-  static Publisher<PublishMetadata> newPublisher(PublisherOptions options) throws ApiException {
+  static Publisher<MessageMetadata> newPublisher(PublisherOptions options) throws ApiException {
     SerializableSupplier<Object> supplier = options.publisherSupplier();
     if (supplier != null) {
       Object supplied = supplier.get();
-      TypeToken<Publisher<PublishMetadata>> token = new TypeToken<Publisher<PublishMetadata>>() {};
+      TypeToken<Publisher<MessageMetadata>> token = new TypeToken<Publisher<MessageMetadata>>() {};
       checkArgument(token.isSupertypeOf(supplied.getClass()));
-      return (Publisher<PublishMetadata>) supplied;
+      return (Publisher<MessageMetadata>) supplied;
     }
-    return new PartitionCountWatchingPublisher(
+
+    TopicPath topic = options.topicPath();
+    PartitionCountWatchingPublisherSettings.Builder publisherSettings =
         PartitionCountWatchingPublisherSettings.newBuilder()
-            .setTopic(options.topicPath())
+            .setTopic(topic)
             .setPublisherFactory(
                 partition ->
                     SinglePartitionPublisherBuilder.newBuilder()
-                        .setTopic(options.topicPath())
+                        .setTopic(topic)
                         .setPartition(partition)
-                        .setContext(PubsubContext.of(FRAMEWORK))
```

Comment:
This needs to be replaced with

```
                  addDefaultMetadata(
                      PubsubContext.of(FRAMEWORK),
                      RoutingMetadata.of(topic, partition),
                      publisher);
```

Reply:
What's the last argument for addDefaultMetadata?

<img width="1028" alt="Screen Shot 2021-04-20 at 10 33 21" src="https://user-images.githubusercontent.com/28604/115414229-db9b1300-a1c3-11eb-9737-ae548c203c2b.png">

Reply:
As per @dpcollins-google , this framework value through setContext is unused in Beam.